libbeat/publisher/queue/diskqueue/writer_loop.go

```diff
@@ -173,6 +173,9 @@ outerLoop:
 				// This can only happen if the queue is being closed; abort.
 				break
 			}
+			// We're creating a new segment file, set the initial bytes written
+			// to the header size.
+			curSegmentResponse.bytesWritten = wl.currentSegment.headerSize()
```

Comment:
This looks like a bug fix. Did we introduce the problem when moving to absolute adresses, or does it fix an issue we have had before? In the later case we might want to add a changelog entry (and maybe a separate PR).

Reply:
No, this was just an oversight in the previous (absolute address) PR, so it hasn't hit a release.